### PR TITLE
Fix multi active subscriptions

### PR DESCRIPTION
**Pull request by biyeun**
add util for better conflict date check
add util to terminate/deactivate active subscriptions when another is being activated.
http://manage.dimagi.com/default.asp?175531#979467

@esoergel @TylerSheffels

### CODE REVIEW

File: corehq/apps/accounting/models.py [hunk 1]
```diff
@@ -901,6 +901,74 @@ def cancel_subscription(self, adjustment_method=None, web_user=None, note=None):
             self, reason=SubscriptionAdjustmentReason.CANCEL, method=adjustment_method, note=note, web_user=web_user,
         )
 
+    def raise_conflicting_dates(self, date_start, date_end):
+        """Raises a subscription Adjustment error if the specified date range
+        conflicts with other subscriptions related to this subscriber.
+        """
+        for sub in Subscription.objects.filter(
+            subscriber=self.subscriber).exclude(id=self.id
+        ).all():
```
Comment (dimagimon) [inline corehq/apps/accounting/models.py]: E124 closing bracket does not match visual indentation

File: corehq/apps/accounting/models.py [hunk 2]
```diff
@@ -901,6 +901,74 @@ def cancel_subscription(self, adjustment_method=None, web_user=None, note=None):
...
+                date_end is not None and date_end > sub.date_start
+            )
+
+            if ((start_before_related_end and start_after_related_start)
+                or (start_after_related_start and related_has_no_end)
+                or (end_after_related_start and end_before_related_end)
+                or (end_after_related_start and related_has_no_end)
+                or (start_before_related_start and end_after_related_end)
+                or (start_before_related_end and current_has_no_end)
+            ):
```
Comment (dimagimon) [inline corehq/apps/accounting/models.py]: E124 closing bracket does not match visual indentation

File: corehq/apps/accounting/models.py [hunk 3]
```diff
@@ -901,6 +901,74 @@ def cancel_subscription(self, adjustment_method=None, web_user=None, note=None):
...
+                raise SubscriptionAdjustmentError(
+                    "The start date of %(start_date)s conflicts with the "
+                    "subscription dates to %(related_sub)s." % {
+                        'start_date': self.date_start.strftime(USER_DATE_FORMAT),
+                        'related_sub': sub,
+                   }
```
Comment (dimagimon) [inline corehq/apps/accounting/models.py]: E122 continuation line missing indentation or outdented

**Comment by czue** [comment]
(might make sense for @proteusvacuum or @nickpell to take a look too since it's billing stuff

**Comment by nickpell** [comment]
@biyeun Do we know that this will fix the misguided errors without breaking anything else by throwing the exceptions?  My instinct is that http://manage.dimagi.com/default.asp?175531#BugEvent.977689 might be a simpler and safer fix

**Comment by biyeun** [comment]
@nickpell updated -- what @czue mentioned worked perfectly, but I left the error checks in place on the billing forms, because that did result in creating overlapping subscriptions without those checks. I removed the more destructive terminate all other subscriptions upon activation date for new subscription process.

**Comment by nickpell** [comment]
:+1: waiting on tests

**Comment by czue** [comment]
glad it worked out!

**Pull request merged**